front/lib/tracking/server.ts

```diff
@@ -2,28 +2,93 @@ import type {
   AgentConfigurationType,
   AgentMessageType,
   DataSourceType,
+  LightWorkspaceType,
+  MembershipRoleType,
   UserMessageType,
   UserType,
   UserTypeWithWorkspaces,
   WorkspaceType,
 } from "@dust-tt/types";
+import * as _ from "lodash";
 
+import { subscriptionForWorkspaces } from "@app/lib/auth";
+import { countActiveSeatsInWorkspace } from "@app/lib/plans/usage/seats";
 import { AmplitudeServerSideTracking } from "@app/lib/tracking/amplitude/server";
+import { CustomerioServerSideTracking } from "@app/lib/tracking/customerio/server";
+import logger from "@app/logger/logger";
 
 export class ServerSideTracking {
   static trackSignup({ user }: { user: UserType }) {
     AmplitudeServerSideTracking.trackSignup({ user });
+    void CustomerioServerSideTracking.trackSignup({ user });
   }
 
-  static async trackUserMemberships({
-    user,
-  }: {
-    user: UserTypeWithWorkspaces;
-  }) {
-    return AmplitudeServerSideTracking.trackUserMemberships({ user });
+  static async trackGetUser({ user }: { user: UserTypeWithWorkspaces }) {
+    try {
+      const subscriptionByWorkspaceId = await subscriptionForWorkspaces(
+        user.workspaces.map((ws) => ws.sId)
+      );
+
+      // TODO(@fontanierh): This is not efficient, we should batch this.
+      const seatsByWorkspaceId = _.keyBy(
+        await Promise.all(
+          user.workspaces.map(async (workspace) => {
+            const seats = await countActiveSeatsInWorkspace(workspace.sId);
```

Comment:
should we use the cached version `countActiveSeatsInWorkspaceCached` ?

Reply:
had no idea this was a thing. The amplitude code used the non-cached version, but I assume cached is fine